webapi/src/main/scala/org/knora/webapi/responders/v1/UsersResponderV1.scala

```diff
@@ -101,57 +91,320 @@ class UsersResponderV1 extends ResponderV1 {
+
+            //_ = log.debug(MessageUtil.toSource(userDataQueryResponse))
+
+            _ = if (userDataQueryResponse.results.bindings.nonEmpty) {
+                throw DuplicateValueException(s"User with the username: '${newUserData.username}' already exists")
             }
 
-            groupIris = groupedUserData.get(OntologyConstants.KnoraBase.IsInGroup) match {
-                case Some(groups) => groups
-                case None => Nil
+            userIri = knoraIdUtil.makeRandomPersonIri
+
+            hashedPassword = BCrypt.hashpw(newUserData.password, BCrypt.gensalt())
+
+            // Create the new user.
+            createNewUserSparqlString = queries.sparql.v1.txt.createNewUser(
+                adminNamedGraphIri = "http://www.knora.org/data/admin",
+                triplestore = settings.triplestoreType,
+                userIri = userIri,
+                userClassIri = OntologyConstants.KnoraBase.User,
+                username = SparqlUtil.any2SparqlLiteral(newUserData.username),
+                password = SparqlUtil.any2SparqlLiteral(hashedPassword),
+                givenName = SparqlUtil.any2SparqlLiteral(newUserData.givenName),
+                familyName = SparqlUtil.any2SparqlLiteral(newUserData.familyName),
+                email = SparqlUtil.any2SparqlLiteral(newUserData.email),
+                preferredLanguage = SparqlUtil.any2SparqlLiteral(newUserData.lang)
+            ).toString
+            //_ = log.debug(s"createNewUser: $createNewUserSparqlString")
+            createResourceResponse <- (storeManager ? SparqlUpdateRequest(createNewUserSparqlString)).mapTo[SparqlUpdateResponse]
+
+
+            // Verify that the user was created.
+            sparqlQuery = queries.sparql.v1.txt.getUserByIri(
+                triplestore = settings.triplestoreType,
+                userIri = userIri
+            ).toString()
+            userDataQueryResponse <- (storeManager ? SparqlSelectRequest(sparqlQuery)).mapTo[SparqlSelectResponse]
+
+            _ = if (userDataQueryResponse.results.bindings.isEmpty) {
+                throw UpdateNotPerformedException(s"User $userIri was not created. Please report this as a possible bug.")
             }
 
-            projectIris: Seq[String] = groupedUserData.get(OntologyConstants.KnoraBase.IsInProject) match {
-                case Some(projects) => projects
-                case None => Nil
+            // create the user profile
+            newUserProfile <- userDataQueryResponse2UserProfile(userDataQueryResponse, UserProfileType.SAFE)
+
+            // create the user operation response
+            userOperationResponseV1 = UserOperationResponseV1(newUserProfile, userProfile.userData)
+
+        } yield userOperationResponseV1
+
+    }
+
+
+    private def updateUserV1(userIri: webapi.IRI, propertyIri: webapi.IRI, newValue: Any, userProfile: UserProfileV1, apiRequestID: UUID): Future[UserOperationResponseV1] = for {
```

Comment:
I've opened an issue for this. Not a quick fix. Also, I need to implement the same thing for groups and projects, so I will redesign it then.